File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> s = new HashSet<>();
        for (int n : nums) s.add(n);
        int ans = 0;
        for (int i = 0; i < nums.length; i ++) {
            int n = nums[i];
            if (s.contains(n-1)) continue;
            int len = 1;
            while (s.contains(n+1)) {
                len ++;
                n++;
            }
            ans = Math.max(ans, len);
        }
        return ans;
    }
}

File: Data Structures & Algorithms/jump-game/submission-1.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        farthest = 0

        for i in range(len(nums)):
            if i > farthest: #如果最远到不了i，就没戏了
                return False
            #根据从当前i开始能跳多远，来更新最远能到哪
            farthest = max(farthest, i+nums[i])
        return True